zad2/warriors.rb

```diff
@@ -1,12 +1,79 @@
-class Warrior
+class Character
+  attr_reader :level
+
   def initialize(name:, level:)
+    unless level.between? @@MIN_LEVEL, @@MAX_LEVEL 
+      raise ArgumentError, "Level should be in range [#@@MIN_LEVEL, #@@MAX_LEVEL]"
+    end
+
+    @name = name
+    @level = level
   end
 
   def strength
+    @level + roll_dice
+  end
+
+  def card
+    "#@name (lvl #@level)"
+  end
+
+  def to_s
+    @name
+  end
+
+  protected
+
+  @@MIN_LEVEL = 1
+  @@MAX_LEVEL = 99
+
+  private
+
+  @@DICE_SIZE = 12
+
+  def roll_dice
+    [*1..@@DICE_SIZE].sample
+  end
+end
+
+class Warrior < Character
+  def level_up!(opponent)
+    level_increase = [0, opponent.level - @level + 1].max
+    new_level = [@level + level_increase, @@MAX_LEVEL].min
+    if new_level > level
```

Comment:
proponuję użyć guard clause

https://www.thechrisoshow.com/2009/02/16/using-guard-clauses-in-your-ruby-code/

Reply:
Poprawione :+1: